tests/cmock/example/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(test)
+
+if(CONFIG_CMOCK)
+  unity_generate_test_runner(src/unit.c)
+  cmock_generate_mocks(${ZEPHYR_BASE}/include/kernel.h .)
```

Comment:
This path seems incorrect and CMake can't find this file. The correct path would be `${ZEPHYR_BASE}/include/zephyr/kernel.h`.

After applying this change (and installing stuff you send in the other thread), I get this error message:
```
ValueError: Unknown template argument kind 350
CMake Error at /workdir/zephyr/subsys/testsuite/cmock/CMakeLists.txt:75 (message):
  1
Call Stack (most recent call first):
  /workdir/zephyr/subsys/testsuite/cmock/CMakeLists.txt:122 (cmock_add_linker_wraps)
  CMakeLists.txt:9 (cmock_generate_mocks)


CMake Error at /workdir/zephyr/subsys/testsuite/cmock/CMakeLists.txt:76 (message):
  Error generating function list from /workdir/zephyr/build/mocks/./kernel.h
Call Stack (most recent call first):
  /workdir/zephyr/subsys/testsuite/cmock/CMakeLists.txt:122 (cmock_add_linker_wraps)
  CMakeLists.txt:9 (cmock_generate_mocks)


-- Configuring incomplete, errors occurred!
See also "/workdir/zephyr/build/CMakeFiles/CMakeOutput.log".
```
I can confirm that `zephyr/build/mocks/./kernel.h` exists and is full of kernel symbols and macros, so I guess that the clang part worked correctly.

Reply:
This is because it was originally written for older version of zephyr before the include directory change. The path now is "zephyr/include/zephyr/kernel.h"